bdb/odh.c

```diff
@@ -1377,3 +1364,45 @@ inline void bdb_cleanup_fld_hints(bdb_state_type *bdb_state)
         bdb_state->fld_hints = NULL;
     }
 }
+
+int bdb_unpack_heap(bdb_state_type *bdb_state, void *in, size_t inlen,
+                    void **out, size_t *outlen, void **freeptr)
+{
+    struct odh odh;
+    int rc;
+    char odhd = bdb_state->ondisk_header;
+
+    /* Force ODH in case that a schema change
+       is removing the ODH from the table. */
+    bdb_state->ondisk_header = 1;
```

Comment:
Reviewed the potential race issue here in person with @riverszhang89 . Otherwise, looks good in general.